server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -685,9 +700,13 @@ private void validateExistingIndex(
             // Index exist - checking that it's closed
             if (currentIndexMetadata.getState() != IndexMetadata.State.CLOSE) {
                 // TODO: Enable restore for open indices
-                throw new SnapshotRestoreException(snapshot, "cannot restore index [" + renamedIndex + "] because an open index " +
-                                                                "with same name already exists in the cluster. Either close or delete the existing index or restore the " +
-                                                                "index under a different name by providing a rename pattern and replacement name");
+                IndexParts indexParts = new IndexParts(renamedIndex);
+                RelationName relationName = new RelationName(indexParts.getSchema(), indexParts.getTable());
+                if (indexParts.isPartitioned()) {
+                    throw new PartitionAlreadyExistsException(new PartitionName(relationName, indexParts.getPartitionIdent()));
+                } else {
+                    throw new RelationAlreadyExists(relationName);
+                }
```

Comment:
Before we had such check in the planner. Now it's irrelevant in planner since we resolve indices in cluster state and also can have rename applied.

Question: apparently it's possible to restore into existing and closed tables (see error message of the ES exception). 

Should we follow up on this as a new feature (with change entry and some test coverage) or let's keep it hidden?